shared/tests/CMakeLists.txt

```diff
@@ -2,7 +2,7 @@
 file(GLOB_RECURSE test_src "*/**.cpp" "*/**.hpp")
 
 if(test_src)
-  add_executable(TestShared "${CMAKE_SOURCE_DIR}/${NUCLEAR_SHARED_DIR}/tests/test.cpp" ${test_src})
+  add_executable(TestShared "${CMAKE_SOURCE_DIR}/shared/tests/test.cpp" ${test_src})
```

Comment:
Using `${NUCLEAR_SHARED_DIR}` is actually more correct than using `shared`

Reply:
If it's not working with that it implies that you are doing things in the wrong order